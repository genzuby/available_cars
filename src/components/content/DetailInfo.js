// @flow
import React from 'react';
import type { Node } from 'react';
import { DETAIL_INFO } from '../../style/componentStyle';

function DetailInfo({ title, desc }: { title: string, desc: string }): Node {
  return (
    <DETAIL_INFO data-testid="detailInfo">
      <h1>{title}</h1>
      <h3>{desc}</h3>
      <p>
        This car is currently available and can be delivered as soon as tomorrow
        morning. Please be aware that delivery times shown in this page are not
        definitive and may change due to bad weather conditions.
      </p>
    </DETAIL_INFO>
  );
}

export default DetailInfo;
