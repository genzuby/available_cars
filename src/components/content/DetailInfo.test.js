import { render } from '@testing-library/react';
import DetailInfo from './DetailInfo';

it('render snapshot', () => {
  const { getByTestId } = render(<DetailInfo />);
  expect(getByTestId('detailInfo')).toMatchSnapshot();
});

it('shows props correctly', () => {
  const { getByTestId } = render(<DetailInfo title={'test'} desc={'desc'} />);
  expect(getByTestId('detailInfo')).toHaveTextContent('test');
  expect(getByTestId('detailInfo')).toHaveTextContent('desc');
});
