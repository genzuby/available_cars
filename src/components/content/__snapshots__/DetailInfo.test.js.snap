// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render snapshot 1`] = `
<div
  class="sc-hBEYos gxzDUi"
  data-testid="detailInfo"
>
  <h1 />
  <h3 />
  <p>
    This car is currently available and can be delivered as soon as tomorrow morning. Please be aware that delivery times shown in this page are not definitive and may change due to bad weather conditions.
  </p>
</div>
`;
